hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -393,14 +454,35 @@ newSn Environment{party} parameters CoordinatedHeadState{confirmedSnapshot, seen
 
 emitSnapshot :: IsTx tx => Environment -> [Effect tx] -> HeadState tx -> (HeadState tx, [Effect tx])
 emitSnapshot env@Environment{party} effects = \case
-  st@OpenState{parameters, coordinatedHeadState} ->
+  st@OpenState{parameters, coordinatedHeadState, previousState} ->
     case newSn env parameters coordinatedHeadState of
       ShouldSnapshot sn txs ->
         ( OpenState
             { parameters
             , coordinatedHeadState = coordinatedHeadState{seenSnapshot = RequestedSnapshot}
+            , previousState
             }
         , NetworkEffect (ReqSn party sn txs) : effects
         )
       _ -> (st, effects)
   st -> (st, effects)
+
+rollback :: HasCallStack => Word -> HeadState tx -> HeadState tx
+rollback n
```

Comment:
So this seems to rely on the fact that `previousState` is updated on every chain transaction seen in sequence. This means also, that the number present here needs to correspond to the number of state transitions.. but how would the chain layer know what how many transitioning events it emited to the head logic?

Just writing down my thoughts as you are probably working on this right now and I'm interested in seeing how you guys solve this :)

Reply:
This function needs some documentation. It should explain or point to the other function which comes up with the `Word` of steps to rollback